plugins/woocommerce/tests/e2e-pw/tests/smoke-tests/upload-plugin.spec.js

```diff
@@ -1,91 +1,56 @@
 const {
 	ADMINSTATE,
-	ADMIN_USER,
-	ADMIN_PASSWORD,
 	GITHUB_TOKEN,
 	PLUGIN_NAME,
 	PLUGIN_REPOSITORY,
 } = process.env;
 const { test, expect } = require( '@playwright/test' );
+const { admin } = require( '../../test-data/data' );
 const path = require( 'path' );
 const {
-	createPlugin,
 	deletePlugin,
-	downloadZip,
 	deleteZip,
-	getLatestReleaseZipUrl,
+	downloadZip,
+	installPluginThruWpCli,
 } = require( '../../utils/plugin-utils' );
 
-const adminUsername = ADMIN_USER ?? 'admin';
-const adminPassword = ADMIN_PASSWORD ?? 'password';
-
-let pluginZipPath;
+let pluginPath;
 let pluginSlug;
 
 test.describe( `${ PLUGIN_NAME } plugin can be uploaded and activated`, () => {
-	// Skip test if PLUGIN_REPOSITORY is falsy.
 	test.skip(
 		! PLUGIN_REPOSITORY,
-		`Skipping this test because value of PLUGIN_REPOSITORY was falsy: ${ PLUGIN_REPOSITORY }`
+		`SKIPPED: PLUGIN_REPOSITORY is falsy: ${ PLUGIN_REPOSITORY }`
 	);
 
 	test.use( { storageState: ADMINSTATE } );
 
-	test.beforeAll( async () => {
-		pluginSlug = PLUGIN_REPOSITORY.split( '/' ).pop();
+	test.beforeAll( async ( { playwright, baseURL } ) => {
+		pluginSlug = path.basename( PLUGIN_REPOSITORY );
 
-		// Get the download URL and filename of the plugin
-		const pluginDownloadURL = await getLatestReleaseZipUrl( {
+		// Download plugin.
+		pluginPath = await downloadZip( {
 			repository: PLUGIN_REPOSITORY,
 			authorizationToken: GITHUB_TOKEN,
 		} );
-		const zipFilename = pluginDownloadURL.split( '/' ).pop();
-		pluginZipPath = path.resolve( __dirname, `../../tmp/${ zipFilename }` );
 
-		// Download the needed plugin.
-		await downloadZip( {
-			url: pluginDownloadURL,
-			downloadPath: pluginZipPath,
-			authToken: GITHUB_TOKEN,
-		} );
-	} );
-
-	test.afterAll( async ( { baseURL, playwright } ) => {
-		// Delete the downloaded zip.
-		await deleteZip( pluginZipPath );
-
-		// Delete the plugin from the test site.
+		// Delete plugin from test site if it's installed.
 		await deletePlugin( {
 			request: playwright.request,
 			baseURL,
 			slug: pluginSlug,
-			username: adminUsername,
-			password: adminPassword,
+			username: admin.username,
+			password: admin.password,
 		} );
 	} );
 
-	test( `can upload and activate ${ PLUGIN_NAME }`, async ( {
-		page,
-		playwright,
-		baseURL,
-	} ) => {
-		// Delete the plugin if it's installed.
-		await deletePlugin( {
-			request: playwright.request,
-			baseURL,
-			slug: pluginSlug,
-			username: adminUsername,
-			password: adminPassword,
-		} );
+	test.afterAll( async ( {} ) => {
+		// Delete the downloaded zip.
+		await deleteZip( pluginPath );
+	} );
 
-		// Install and activate plugin
-		await createPlugin( {
-			request: playwright.request,
-			baseURL,
-			slug: pluginSlug.split( '/' ).pop(),
-			username: adminUsername,
-			password: adminPassword,
-		} );
+	test( `can upload and activate ${ PLUGIN_NAME }`, async ( { page } ) => {
+		await installPluginThruWpCli( pluginPath );
```

Comment:
Plugin will be uploaded through WP CLI commands that will be run inside the `tests-cli` container in WP ENV. This is due to a limitation where wp env couldn't upload plugins through the UI.

Reply:
Do all sites use installPluginThruWpCli? Just thinking about if anyone is using the tests on sites without WP CLI

Reply:
I think I have answered my own question from one of the other comments, we only use this with localhost right now.
"Plugin tests are run on localhost anyway, not on the daily smoke test site"